src/components/PostDetails.tsx

```diff
@@ -1,117 +1,152 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { client } from '../utils/fetchClient';
+
+type Props = {
+  selectedPost: Post | null,
+  isCommentButtonClicked: boolean,
+  setIsCommentButtonClicked: (value: boolean) => void,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+  isCommentButtonClicked,
+  setIsCommentButtonClicked,
+}) => {
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [commentErrorMessage, setCommentErrorMessage] = useState('');
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+
+  const commentButtonHandler = () => {
+    setIsCommentButtonClicked(true);
+  };
+
+  const getPostComments = () => {
+    return client.get<Comment[]>(`/comments?postId=${selectedPost?.id}`)
+      .catch(() => {
+        setCommentErrorMessage('Something went wrong!');
+      });
+  };
+
+  useEffect(() => {
+    setIsCommentsLoading(true);
+    getPostComments()
+      .then(commentsFromServer => {
+        if (commentsFromServer) {
+          setPostComments(commentsFromServer);
+        }
+      })
+      .finally(() => setIsCommentsLoading(false));
+  }, [selectedPost]);
+
+  const deleteComment = (id: number) => {
```

Comment:
I'd move all such functions into a separate file

Reply:
Still not fixed from the previous review